commands/sql/sqlsync.drush.inc

```diff
@@ -250,9 +250,9 @@ function drush_sqlsync_sql_sync($source, $destination) {
   else {
     $tmp = '/tmp'; // Our fallback plan.
     drush_log(dt('Starting to discover temporary files directory on Destination.'), LogLevel::OK);
-    $return = drush_invoke_process($destination, 'php-eval', array('return drush_find_tmp();'), array(), array('integrate' => FALSE, 'override-simulated' => TRUE));
-    if (!$return['error_status']) {
-      $tmp = $return['object'];
+    $return = drush_invoke_process($destination, 'core-status', array(), array(), array('integrate' => FALSE, 'override-simulated' => TRUE));
```

Comment:
This change looks good. Feel free to cherry pick it ahead of the rest of the PR. If you want, you can force-push to this branch after this change is in.